diff --git a/dlls/winemac.drv/cocoa_window.m b/dlls/winemac.drv/cocoa_window.m
index 3e5d2b7..c9f8972 100644
--- a/dlls/winemac.drv/cocoa_window.m
+++ b/dlls/winemac.drv/cocoa_window.m
@@ -982,7 +982,11 @@ @implementation WineMetalView
 
     - (CALayer*) makeBackingLayer
     {
+#if defined(__x86_64__)
         CAMetalLayer *layer = [WineMetalLayer layer];   /* CW HACK 22435 */
+#else
+        CAMetalLayer *layer = [CAMetalLayer layer];
+#endif
         layer.device = _device;
         layer.framebufferOnly = YES;
         layer.magnificationFilter = kCAFilterNearest;
diff --git a/dlls/winemac.drv/d3dmetal.c b/dlls/winemac.drv/d3dmetal.c
index a1119f3..bd0ce73 100644
--- a/dlls/winemac.drv/d3dmetal.c
+++ b/dlls/winemac.drv/d3dmetal.c
@@ -434,4 +434,13 @@ static void cf_client_surface_release(CFAllocatorRef allocator, const void *clie
     client_surface_release((struct client_surface *)client_surface);
 }
 
+#else
+
+#include "config.h"
+#include "macdrv.h"
+
+void macdrv_client_surface_presented(const macdrv_event *event)
+{
+}
+
 #endif
